modules/cloud_provider_equinix_metal/examples/simple/variables.tf

```diff
@@ -0,0 +1,36 @@
+# TEMPLATE: This file was automatically generated with `generate_addon_structure.sh`
+# TEMPLATE: and should be modified as necessary
+# TEMPLATE:
+# TEMPLATE: All variables must have a description and should declare their type.
+# TEMPLATE: Set defaults whenever possible but do not set defaults for required properties.
+# TEMPLATE: Declare all variables in this file, sprawling declarations are difficult to identify.
+# TEMPLATE:
+# TEMPLATE: https://www.terraform.io/docs/language/values/variables.html
+# TEMPLATE: https://www.terraform.io/docs/language/expressions/types.html
+#
+variable "metal_auth_token" {
+  type        = string
```

Comment:
I don't know if these should be top-level modules for the add-ons, or if we should accept the pain of duplication across add-on config?

The MetalLB add-on uses these variables when creating IP resources, but the CPEM add-on does not create any Equinix resources itself.  In the CPEM add-on, the auth token and project ID are used by CPEM after installation to manage Equinix resources at runtime, and they don't necessarily have to be the same token/project that was used to create the k8s cluster.

Reply:
Ah shoot, i just realized I may have made an opinionated choice with my change below to use metal_auth_token everywhere when instead it maybe should be separated to "cluster level" versus "infra level" scopes?

Reply:
I thought about this a bit more, and I think this is ultimately a non-issue.  I was conflating the `examples/simple` module with the add-ons module...but the add-ons module has no auth token variable; that's only in `examples/simple` because of the provider.  This example doesn't even touch the `equinix` provider, so I removed the provider in the hope that it will confuse me less.